third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -79,13 +81,26 @@ std::vector<ComputationClient::DataPtr> PjRtComputationClient::TransferToServer(
   std::vector<ComputationClient::DataPtr> datas;
   datas.reserve(tensors.size());
   for (auto& tensor : tensors) {
-    xla::Literal literal(tensor.shape);
-    tensor.populate_fn(tensor, literal.untyped_data(), literal.size_bytes());
-
     PjRtDevice* pjrt_device = StringToPjRtDevice(tensor.device);
-    std::shared_ptr<xla::PjRtBuffer> buffer =
-        client_->BufferFromHostLiteral(literal, pjrt_device).ValueOrDie();
-    buffer->GetReadyFuture().Await();
+
+    auto literal = std::make_shared<xla::Literal>(tensor.shape);
```

Comment:
why are you adding `make_shared` here?

Reply:
See above for how/why we pass ownership of `literal` to the `on_done_with_host_buffer` callback. To be honest, I couldn't figure out how to make this compile with `unique_ptr` or `literal` as a stack variable, so I used `shared_ptr` and "borrowed" the [logic used in `pjrt_stream_executor_client`](https://github.com/tensorflow/tensorflow/blob/a37ac96c6eefe809003f29a815b915bfed5ad36b/tensorflow/compiler/xla/pjrt/pjrt_stream_executor_client.cc#L1491-L1501).

Not sure if this is the best way. I'm definitely open to alternate ways of writing this.